XenCenterLib/Archive/ArchiveWriter.cs

```diff
@@ -73,61 +73,49 @@ public void CreateArchive(string pathToArchive, Action cancellingDelegate = null
         /// <param name="totalProgressIncrease">Total progress that needs to be added for archiving this directory. In the first recursive call it should be 100. If the folder we're adding should add 18 percentage points to the total progress, set this value to 18.</param>
         /// <param name="progressOffset">Offset to the progress. This is added to <see cref="totalProgressIncrease"/> when setting the progress for this directory. If this folder should add 18 percentage points to the total progress, but it's for a folder past the 50% mark of the total progress (i.e.: completing this folder should set the total to 68), set this value to 50.</param>
         /// <exception cref="FileNotFoundException">A directory could not be found.</exception>
-        private void PopulateArchive(string pathToArchive, string pathToCurrentDirectory, Action cancellingDelegate = null, Action<int> progressDelegate = null, int totalProgressIncrease = 100, int progressOffset = 0)
+        private void PopulateArchive(string pathToArchive, string pathToCurrentDirectory, Action cancellingDelegate = null, Action<int> progressDelegate = null, float totalProgressIncrease = 100, float progressOffset = 0)
         {
             cancellingDelegate?.Invoke();
 
-            var longPathToArchive = StringUtility.ToLongWindowsPath(pathToArchive);
+            pathToArchive = StringUtility.ToLongWindowsPath(pathToArchive, true);
+            if (!Directory.Exists(pathToArchive))
+                throw new FileNotFoundException($"The path {pathToArchive} does not exist");
 
-            if (!Directory.Exists(longPathToArchive))
-                throw new FileNotFoundException("The path " + pathToArchive + " does not exist");
+            pathToCurrentDirectory = StringUtility.ToLongWindowsPath(pathToCurrentDirectory, true);
 
-            AddFiles(longPathToArchive, pathToCurrentDirectory, cancellingDelegate);
+            //add the current directory except when it's the root directory
+            if (pathToArchive != pathToCurrentDirectory)
+                AddDirectory(CleanRelativePathName(pathToArchive, pathToCurrentDirectory), Directory.GetCreationTime(pathToCurrentDirectory));
 
+            var files = Directory.GetFiles(pathToCurrentDirectory);
             var directories = Directory.GetDirectories(pathToCurrentDirectory);
 
-            if (directories.Length == 0)
-            {
-                progressDelegate?.Invoke(totalProgressIncrease + progressOffset);
+            if (files.Length == 0 && directories.Length == 0)

```

Comment:
I also realised I had forgotten this check - it doesn't fail because the increment is float and it results to `Infinity`, but there is no need to proceed if there are no files and dirs.